pkg/operator/controller/garden/garden/reconciler_reconcile.go

```diff
@@ -950,6 +960,74 @@ func (r *Reconciler) updateHelmChartRefForGardenlets(ctx context.Context, log lo
 	return nil
 }
 
+func (r *Reconciler) reconcileDNSRecords(ctx context.Context, log logr.Logger, garden *operatorv1alpha1.Garden) error {
+	if garden.Spec.VirtualCluster.DNS.Provider == nil || garden.Spec.VirtualCluster.DNS.SecretRef == nil {
+		return fmt.Errorf("no DNS provider or DNS secret configuration found in Garden resource")
+	}
+
+	dnsRecordList := &extensionsv1alpha1.DNSRecordList{}
+	if err := r.RuntimeClientSet.Client().List(ctx, dnsRecordList, client.InNamespace(r.GardenNamespace)); err != nil {
+		return fmt.Errorf("failed listing DNS records: %w", err)
+	}
+
+	staleDNSRecordNames := sets.New[string]()
+	for _, dnsRecord := range dnsRecordList.Items {
+		staleDNSRecordNames.Insert(dnsRecord.Name)
+	}
+
+	istioIngressGatewayLoadBalancerAddress, err := kubernetesutils.WaitUntilLoadBalancerIsReady(ctx, log, r.RuntimeClientSet.Client(), namePrefix+v1beta1constants.DefaultSNIIngressNamespace, v1beta1constants.DefaultSNIIngressServiceName, time.Minute)
```

Comment:
Isn't that already covered/guaranteed by the `deployIstio` step?

Reply:
`deployIstio` deploys the chart containing the service of type `LoadBalancer` via a managed resource. As far as I understand the MR health check, it checks the load balancer status, too.
Nonetheless, the code looks appropriate for me, as the load balancer address needs to be looked up anyway.